Sources/SwifterSwift/SwiftStdlib/DecodableExtensions.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  DecodableExtensions.swift
+//  SwifterSwift
+//
+//  Created by Mustafa GUNES on 16.01.2020.
+//  Copyright © 2020 SwifterSwift
+//
+
+#if canImport(Foundation)
+import Foundation
+#endif
+
+public extension Decodable {
+
+    #if canImport(Foundation)
+    /// SwifterSwift: Parsing the model in Codable type
+    /// - Parameter data: Data.
+    /// - Returns: Decoded self model value.
+    /// - Throws: Decoding error.
+    static func decode(_ data: Data) throws -> Self? {
+        let decoder = JSONDecoder()
+        decoder.dataDecodingStrategy = .deferredToData
+        do {
+            let parsed = try decoder.decode(self, from: data)
+            return parsed
+        } catch {
+            return nil
+        }
+    }
```

Comment:
@mustafagunes Thanks for your extension, here is my two cents on it:

1. The method is marked as `throws`, but you're ignoring the error, but not throwing the decoding error anyway!
2. What happens if implementers have a custom `JSONDecoder`, or different than the one used in this method?
3. Why are you using `dataDecodingStrategy` = `.deferredToData`?
4. I think this makes sense to be a failable initializer instead of a static function

```swift
init?(from data: Data, using decoder: JSONDecoder = .init()) {
    guard let parsed = try? decoder.decode(Self.self, from: data) else { return nil }
    self = parsed
}
```